### DISCUSSION TITLE
Lesson 6: error when compiling MockV3Aggregator

### DISCUSSION THREAD

**Discussion by vjumminz** [smartcontractkit/full-blockchain-solidity-course-py#78] [Q&A] [answered]
Hi guys, 

I'm getting an error when trying to compile the MockV3Aggregator.sol contract. The error looks like this:
![Untitled](https://user-images.githubusercontent.com/24928833/134509982-3b374a79-284d-47d8-80a6-fc917490bcca.png)

The only thing I did was copy the code from https://github.com/smartcontractkit/chainlink-mix/blob/master/contracts/test/MockV3Aggregator.sol as instructed. Any reason why this might be occurring?

Appreciate the help and really enjoying the course so far :)

**Comment by PatrickAlphaC** [comment, marked as answer]
Good question! 

It looks like you’re trying to have an interface inherit an interface. What if you make this a contract instead of an interface?

**Reply by vjumminz** [reply]
Appreciate the response -- all it needed was to change solidity versions! 

Thank you Patrick for all your effort :)

**Reply by romanwiseman** [reply]
Hey @vincecummins , could I ask what did you change your solidity version to? And did you have to change on all contracts I presume?

**Reply by CoraCote** [reply]
Appreciate the response, Thanks.

**Comment by romanwiseman** [comment]
Hey,

I'm having the exact same issue on lesson 6 after copy/pasting the MockV3Aggregator.sol code from the chainlink-mix github.

This contract inherits from AggregatorV2V3Interface.sol which in turn is trying to inherit from AggregatorInterface and AggregatorV3Interace. 

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.6.0;

import "./AggregatorInterface.sol";
import "./AggregatorV3Interface.sol";

interface AggregatorV2V3Interface is AggregatorInterface, AggregatorV3Interface
{
}
```

I tried changing this file to a contract rather than an interface as suggested but it then threw an error saying it needed to be marked as abstract - tried adding a few keyword combinations including abstract and public but to no avail!

I've just checked and all my code is using solidity ^0.6.0 so there doesn't seem to be any conflict there.

Any help would be much appreciated!

(fyi not sure how to comment in here but didn't want to create a duplicate thread... this is obviously not an answer!)

**Reply by vjumminz** [reply]
Something else I did -- I cloned Patrick's brownie_fund_me repository and then used `brownie compile` to compile his smart contracts. In the process of this, I believe solc was reinstalled (?) which took approximately 5 mins and it showed a progress bar. Not sure if this will help but as soon as I tried to compile my contracts the error was gone.

I changed my solidity version to 0.6.6 but from your code this wouldn't appear to be the issue.

Hope this helps

**Reply by romanwiseman** [reply]
Awesome, I did that and it installed another solc version and then compiled fine!

Thanks

**Reply by msperry2014** [reply]
This also worked for me.  It looks like it downloaded a local copy of the solc compiler (v0.6.12).  Thanks!

**Reply by msperry2014** [reply]
However, I now get a new error when going to run the deploy.py script:
`Brownie v1.16.4 - Python development framework for Ethereum

  File "C:\Users\Mike, line line, in in
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "C:\Users\Mike, line line, in in
    active_project.load_config()
  File "C:\Users\Mike, line line, in in
    _load_project_config(self._path)
  File "C:\Users\Mike, line line, in in
    and "cmd_settings" in values
TypeError: argument of type 'NoneType' is not iterable
PS D:\Documents\Projects\BlockChain\brownie_fund_me> `

It's interesting that the version that I cloned in order to get the solc compiler runs the script just fine.  

It looks like I broke something basic.  Does anyone have any ideas?

Python 3.9.7, ganache-cli v6.12.2 (ganache-core 2.13.2), brownie 1.16.4

**Reply by PatrickAlphaC** [reply]
Could you make this in a new question with as much information as possible? (what command you ran, what the scripts look like, any contracts)

**Reply by msperry2014** [reply]
> Could you make this in a new question with as much information as possible? (what command you ran, what the scripts look like, any contracts)

I found the problem, I copied FundMe.sol from the wrong repository:
Incorrect: https://github.com/PatrickAlphaC/brownie_fund_me/blob/main/contracts/FundMe.sol
Correct: https://github.com/PatrickAlphaC/fund_me/blob/main/FundMe.sol

  The difference is in the constructor, the pricefeed parameter must be supplied (I'm assuming later in the lesson).

  Thanks for your help Patrick, these lessons are awesome!

**Reply by andremiracle** [reply]
hey guys i was having same issue and i did clone the repository compiled the smart contracts and all that still nothing happened

**Reply by cromewar** [reply]
Hello @andremiracle did you change the solidity info? if nothing works please create a new question with more detailed info.

**Reply by andremiracle** [reply]
The whole issue was from my yaml file which I've fixed. thanks for reaching out

**Reply by gigu007** [reply]
great it worked for me

**Comment by 0xZuk0** [comment]
I was having the same issue i fixed it by changing solidity version from ^0.6.0 to ^0.7.0. I also tried with ^0.8.0 but it started to give me another error
"TypeError: "send" and "transfer" are only available for objects of type "address payable", not "address""

So i guess just change it to ^0.7.0 and it should work

**Reply by PatrickAlphaC** [reply]
Version 0.8 of solidity has a few breaking changes. Interesting that moving to 0.7 worked... at first glance... I don't think that should... but if it works that's great!

**Reply by engfuentes** [reply]
thanks, this solution worked for me. 

I changed it to >=0.7.0. and then it downloaded solc 0.8.9 and compiled 

solc 0.8.9 successfully installed at: C:\Users\rodri\.solcx\solc-v0.8.9\solc.exe
Compiling contracts...
  Solc version: 0.8.9
  Optimizer: Enabled  Runs: 200
  EVM Version: Istanbul
Generating build data...
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorInterface
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorV2V3Interface
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorV3Interface
 - MockV3Aggregator

**Reply by techwithnaga** [reply]
Excellent! this solution works for me. When you change solidity version from 0.6.0 to 0.7.0 and compile it, It downloads a new solc version. 0.7.6. For whatever reason, solc 0.7.6 compiles MockV3Aggregator.

**Reply by shlok-codes** [reply]
Worked for me too.

I changed to ^0.7.0. And it downloaded Solc version: 0.7.6 and compiled.

**Reply by CoraCote** [reply]
Perfect! This solution is good for me.
